mmaction/datasets/pipelines/loading.py

```diff
@@ -814,6 +814,87 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class PyAVDecodeMotionVector(PyAVDecode):
+    """Using pyav to decode the motion vectors from video.
+
+    PyAV: https://github.com/mikeboers/PyAV
+
+    Required keys are "video_reader" and "frame_inds",
+    added or modified keys are "motion_vectors".
+
+    Args:
+        multi_thread (bool): If set to True, it will apply multi
+            thread processing. Default: False.
+    """
+
+    def _parse_vectors(self, mv, vectors, height, width):
+        """Parse the returned vectors."""
+        (w, h, src_x, src_y, dst_x,
+         dst_y) = (vectors['w'], vectors['h'], vectors['src_x'],
+                   vectors['src_y'], vectors['dst_x'], vectors['dst_y'])
+        val_x = dst_x - src_x
+        val_y = dst_y - src_y
+        start_x = (-1 * w / 2).astype(np.int8) + dst_x
+        start_y = (-1 * h / 2).astype(np.int8) + dst_y
+        end_x = start_x + w.astype(np.int8)
+        end_y = start_y + h.astype(np.int8)
+        for row, _ in enumerate(vectors):
+            if (start_x[row] >= 0 and end_x[row] < width and start_y[row] >= 0
+                    and end_y[row] < height):
+                mv[start_y[row]:end_y[row],
+                   start_x[row]:end_x[row]] = (val_x[row], val_y[row])
+
+        return mv
+
+    def __call__(self, results):
+        """Perform the PyAV motion vector decoding.
+
+        Args:
+            results (dict): The resulting dict to be modified and passed
+                to the next transform in pipeline.
+        """
+        container = results['video_reader']
+        imgs = list()
+
+        if self.multi_thread:
+            container.streams.video[0].thread_type = 'AUTO'
+        if results['frame_inds'].ndim != 1:
+            results['frame_inds'] = np.squeeze(results['frame_inds'])
+
+        # set max indice to make early stop
```

Comment:
`indice` -> `index`

Reply:
Only if  size of 'frame_inds' is like (1, 32) -> (32). It is the same as `PyAVDecode`.

Reply:
yeah so frame_inds belongs to modified keys?

Reply:
ok